formal-spec/Leios/SimpleSpec.agda

```diff
@@ -115,45 +121,62 @@ data _⇀⟦_⟧_ : Maybe LeiosState → LeiosInput → LeiosState × LeiosOutpu
        ────────────────────────────────────
        just s ⇀⟦ FTCH-LDG ⟧ (s , FTCH-LDG l)
 
-  -- TODO: Base chain
+  -- Base chain
+  --
+  -- Note: Submitted data to the base chain is only taken into account
+  --       if the party submitting is the block producer on the base chain
+  --       for the given slot
+
+  Base₁ : ∀ {txs} → let open LeiosState s in
+        ──────────────────────────────────────────────────────────────────
+        just s ⇀⟦ SUBMIT (inj₂ txs) ⟧ (record s { MemPool = txs } , EMPTY)
```

Comment:
It's an interesting question what to do here. `SUBMIT` for the base chain has this kind of one time semantics and does not store anything, and doing it like this has a similar flavour. So for consistency this makes sense, but then we'd have to do a `SUBMIT` to Leios containing the most recent mempool every slot, which I'm not sure is what the paper intended. But now that I reread it, I think the paper is just unclear here, and it could be either option.

Reply:
Though I think if we keep this overwrite semantics we should probably rename `MemPool` because that's misleading. The only thing that comes to mind right now is `ToPropose`, which I think is also bad, but less so...

Reply:
In _Figure 2: The Simplified Leios Protocol_ txs are the most recently stored transactions - therefore the overwrite semantics

Reply:
My initial understanding was that if it says 'let txs be the (most recently stored) transactions', it means that from the stored transactions you select a subset and then you'd have the append semantics. I think either interpretation is consistent with the paper, but this one is consistent with the base chain as well.